### PR TITLE
Add CUDA 9.2 support to 0.3.2 release

### PR DESCRIPTION
**Pull request by BenjaminW3**
* add CUDA 9.2 support #533
* update Readme #530, #542, #536
* update CMake checks #528

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
Could you please add the PR numbers which you are currently back-port that we have a better overview and move around the cards in the the corresponding project.

**Comment by BenjaminW3** [comment]
Done. This is still work in progress as I have to get the CI green for this huge build matrix.

**Comment by BenjaminW3** [comment]
Short progress update: More and more jobs are getting green... Still some CUDA / clang issues to fix.

**Comment by psychocoderHPC** [comment]
As I can see a few tests with clang 3.5.2 tries to use CUDA.

**Comment by BenjaminW3** [comment]
Yes, I will have to change the .travis.yml file a bit. I do not think that there are any other code changes necessary anymore to get CUDA 9.2 working.

**Comment by BenjaminW3** [comment]
Finally it is green!

File: alpakaConfig.cmake [hunk 1]
```diff
@@ -357,6 +379,46 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
                 ENDIF()
 
             ELSE()
+                IF("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+                    IF(CUDA_VERSION VERSION_EQUAL 8.0)
+                        IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 5.0)
```
Comment (psychocoderHPC) [inline alpakaConfig.cmake]: I was not aware that we have the gcc guard also added to alpaka and not only to PIConGPU.
GCC 5.3 and CUDA 8 are compatible (https://github.com/ComputationalRadiationPhysics/picongpu/pull/2636)
I will add a PR to alpaka to fix it.

Comment (ax3l) [reply alpakaConfig.cmake]: Added in #566

**Pull request merged**